src/gg.regression.unity.bots/Runtime/Scripts/RGHeadlessSequenceRunner.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using System.Reflection;
+using RegressionGames.StateRecorder.BotSegments;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using RegressionGames.Types;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using Object = UnityEngine.Object;
+
+// ReSharper disable InconsistentNaming
+namespace RegressionGames
+{
+    public static class RGHeadlessSequenceRunner
+    {
+
+        internal static readonly int Rc_RunSuccess = 0;
+        internal static readonly int Rc_RunFailed = 1;
+        internal static readonly int Rc_SequencePathMissing = 2;
+        internal static readonly int Rc_OverlayNotInScene = 3;
+        internal static readonly int Rc_SequenceLoadFailure = 4;
+        internal static readonly int Rc_SequenceTimeoutNeedsPath = 5;
+        internal static readonly int Rc_SequenceTimeoutMissing = 6;
+        internal static readonly int Rc_SequenceTimeoutNotInt = 7;
+
+        internal static readonly string SequencePathArgument = "-rgsequencepath";
+        internal static readonly string SequenceTimeoutArgument = "-rgsequencetimeout";
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSplashScreen)]
+        public static void FirstLoadChecks()
+        {
+            // do this here to fail fast on bad args
+            ParsePathArgument();
+            ParseTimeoutArgument();
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterSceneLoad)]
+        public static void Initialize()
+        {
+            var resourcePath = ParsePathArgument();
+            if (resourcePath != null)
+            {
+                BotSegmentsPlaybackController playbackController = Object.FindObjectOfType<BotSegmentsPlaybackController>();
+                if (playbackController == null)
+                {
+                    RGDebug.LogError($"RGOverlay must be in the first scene loaded for your game to utilize {SequencePathArgument} command line argument");
+                    Application.Quit(Rc_OverlayNotInScene);
+                }
+
+                playbackController.gameObject.AddComponent<RGHeadlessSequenceRunnerBehaviour>();
+
+                //ensure a keyboard exists for the input playback
+                if (IsBatchMode())
+                {
+                    var currentKeyboard = Keyboard.current;
```

Comment:
If they already have a keyboard would we also need to attempt to set flags?

Reply:
We definitely do not today as we don't create a virtual keyboard for our runs outside of this headless environment.

It's a good philosophical discussion we should experiment with though in the future